File: Kirin/core/wp8/common/javascript/src/Zomg.js
defineModule("Zomg", function (require, exports) {
	var module;
	exports.onLoad = function(obj) {
		console.log("onLoad done!", "INFO");
		module = obj;
		obj.Arrrgh();
		obj.HeresAString_("a string");
		obj.HeresAnInt_(1337);
		obj.HeresABool_(true);
		obj.HeresAStringArray_(["string a", "string b", "string c", "string d"]);
		obj.HeresAnIntArray_([17, 123, 345, 823, 11, 99]);
		obj.HeresABoolArray_([true, false, false, false, true]);
		obj.HeresSomeArrays___(["arr1", "arr2", "arr3"], [1, 2, 3], [false, true, false]);
		//obj.HeresSomeArraysOfArrays__([["string 00", "string 01"], ["string 10", "string 11", "string 12"]], [[[0, 1, 2], [2, 3 4]], [[1, 2, 3], [4, 5 6]]]);
		
		var config = {};
		config.method = "GET";
		config.url = "http://www.google.co.uk/";
		config.postData = null;
		config.headers = "";
		config.payload = function(payload) {
			console.log("THE PAYLOAD WAS " + payload);
		};
		config.onError = function(err) {
			console.log("THE ERROR WAS " + err);
		};
		require("Networking").downloadString(config);
		
		var settings = require("Settings");
		settings.put("when i see you celtic", "i go out of my head");
		settings.put("i just can't get enough", "i just can't get enough");
		settings.put("all the things you do to me", "and everything you say");
		settings.put("i slip and slide and i fall in love", "and i just can't seem to get enough of");
		settings.commit();
		//settings.put("oh", "yes");
		//settings.commit();
		console.log("oh: " + settings.get("oh"));
	};
	
	exports.Whateva = function(num) {
		console.log("Whateva " + num);
	};
});

File: src/wp8/KirinWP8Test/Zomg.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text; 
using System.Threading.Tasks;
 
namespace KirinTest
{ 
    class Zomg 
    { 
        public void Arrrgh()
        { 
            Debug.WriteLine("it's arrrgh!"); 
        }

        public void HeresAString(string str)
        {
            Debug.WriteLine(str);
        }

        public void HeresAnInt(int i)
        {
            Debug.WriteLine(i);
        }

        public void HeresABool(bool b)
        {
            Debug.WriteLine(b);
        }

        public void HeresAStringArray(string[] strs)
        {
            Array.ForEach(strs, HeresAString);
        }
         
        public void HeresAnIntArray(int[] ints)
        {
            Array.ForEach(ints, HeresAnInt);
        }

        public void HeresABoolArray(bool[] bools)
        {
            Array.ForEach(bools, HeresABool);
        }

        public void HeresSomeArrays(string[] strs, int[] ints, bool[] bools)
        {
            HeresAStringArray(strs);
            HeresAnIntArray(ints);
            HeresABoolArray(bools);
        }

        /*public void HeresSomeArraysOfArrays(string[][] strs, int[][][] ints)
        {
            Array.ForEach(strs, HeresAStringArray);
            Array.ForEach(ints, (x) => Array.ForEach(x, HeresAnIntArray));
        }*/
    } 
} 
